src/lsp/cobol_parser/text_tokenizer.ml

```diff
@@ -244,13 +467,15 @@ let preproc_n_combine_tokens ~source_format =
     | [CONSTANT]                     -> Error `MissingInputs
     | CONSTANT :: RECORD :: _          -> subst_n CONSTANT_RECORD 2
 
-    | [PROGRAM_ID]
-    | [PROGRAM_ID; PERIOD]           -> Error `MissingInputs
-    | PROGRAM_ID :: PERIOD :: _        -> info_word_after 2
-    | PROGRAM_ID :: _                  -> info_word_after 1
+    | [(PROGRAM_ID | FUNCTION_ID)]
```

Comment:
For readability of each case, please duplicate entries instead of using parentheses here.